launcher/ui/dialogs/ProgressDialog.cpp

```diff
@@ -1,29 +1,69 @@
-/* Copyright 2013-2021 MultiMC Contributors
+/// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  PrismLaucher - Minecraft Launcher
+ *  Copyright (C) 2023 Rachel Powers <508861+Ryex@users.noreply.github.com>
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * This file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      Copyright 2013-2021 MultiMC Contributors
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
  */
 
 #include "ProgressDialog.h"
 #include "ui_ProgressDialog.h"
 
+#include <limits>
 #include <QDebug>
 #include <QKeyEvent>
 
 #include "tasks/Task.h"
 
+#include "ui/widgets/SubTaskProgressBar.h"
+
+
+// map a value in a numaric range of an arbatray type to between 0 and INT_MAX
+// for getting the best precision out of the qt progress bar
```

Comment:
```suggestion
// map a value in a numeric range of an arbitrary type to between 0 and INT_MAX
// for getting the best precision out of the qt progress bar
```

does it even make sense though? how are you getting more precision from the thing, the only thing I see happening is double precision adding a bit of variability to the end value, but that's not precision, that's just noise o.O

Reply:
Technically this could be accomplished by mapping to a much smaller range because the smallest unit it can draw is one pixel width and the mad number of those across a screen is going to be much less than 4000

the issue is that normal type coercion will just lose data with types larger than `INT_MAX`.
Also, for the progress bar it's somewhat important to use a consistent range size so that it doesn't jump around, and looks smooth as it progresses next to other bars.